src/f32/spirv.rs

```diff
@@ -0,0 +1,88 @@
+pub trait MathExt {
+    fn pow(self, factor: Self) -> Self;
+    fn sqrt(self) -> Self;
+    fn log2(self) -> Self;
+    fn abs(self) -> Self;
+    fn cos(self) -> Self;
+    fn sin(self) -> Self;
+    fn sin_cos(self) -> (f32, f32);
+    fn round(self) -> Self;
+    fn floor(self) -> Self;
+    fn ceil(self) -> Self;
+    fn exp(self) -> Self;
+    fn saturate(self) -> Self;
+
+    fn signum(self) -> Self;
+    fn copysign(self, sign: Self) -> Self;
+
+    // TODO: implement but need new intrinsic in core?
+    //fn tan(self) -> Self;
```

Comment:
LLVM doesn't have a `tan` intrinsic, unlike all the other methods here.

Reply:
There's a `tanf` implementation here if you need one https://github.com/rust-lang/libm/blob/master/src/math/tanf.rs.